src/Control/Session.php

```diff
@@ -307,7 +312,16 @@ public function start(HTTPRequest $request)
         $data = [];
         if (!session_id() && (!headers_sent() || $this->requestContainsSessionId($request))) {
             if (!headers_sent()) {
-                session_set_cookie_params($timeout ?: 0, $path, $domain ?: null, $secure, true);
+                $sameSite = static::config()->get('cookie_samesite');
+                $secure = $this->isCookieSecure($sameSite, Director::is_https($request));
```

Comment:
Will `Director::is_https($request)` handle the scenario where https is terminated at a load balancer?

Reply:
I have no idea. But changing that behaviour is out of scope for this PR, I'd say.